cmd/serve.go

```diff
@@ -49,7 +48,7 @@ func StartServer(logger *log.Zap, cfg *config.Shield) error {
 	}
 
 	// @TODO: need to inject custom logger wrapper over zap into ctx to use it internally
-	ctx, cancelFunc := context.WithCancel(salt_server.HandleSignals(context.Background()))
+	ctx, cancelFunc := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
os.Interrupt is same as syscall.SIGINT   
os.Kill is same as syscall.SIGKILL
(Refer https://pkg.go.dev/os#Signal)

Do not trap SIGKILL signal. only trap SIGTERM and SIGINT

Reply:
My understanding of this: If we would trap SIGKILL, it will kill the process and hence the server will not shut gracefully. CMIIW.

Reply:
Trapping sigkill is pointless for 2 reasons:

1. SIGKILL is usually issued by us or the system when the process is not responding for some reason. Under those circumstances, it's fair to assume that the graceful shutdown logic will also may not work as expected.
2. On issuing a sigkill, kernel will kill the process non-gracefully anyway (there is no guarantee that the graceful shutdown logic will be allowed to execute)